examples/platforms/dart/test/ExternalTypes_test.dart

```diff
@@ -94,4 +99,9 @@ void main() {
     expect(result.color, struct.color);
     expect(result.season, struct.season);
   });
+  _testSuite.test("Use MyClass", () {
+    final result = UseMyClass.callFoo(MyDartClass());
```

Comment:
almost, what I meant is, here you should call bar(), and result should still be 77.
C++'s bar() then calls C++ foo(), but since foo() is virtual, and overriden in MyDartClass, the result is 77

Reply:
also, would it be possible to have a more OOP way, like:

```
var myclass = UseMyClass();
myclass.bar();
```

without the static methods

Reply:
> but since foo() is virtual, and overriden in MyDartClass, the result is 77

Technically it's problematic without generation of class which derives from MyClass in C++.
Please describe a bit more your use case, probably it might be solved somehow differently.

Reply:
@iamsergio, I've updated the example according to your comments.

@Hsilgos, I think you're over-complicating things. Using a pre-existing non-abstract class as an "external" interface in IDL just works, as this example clearly shows.

Reply:
```
class MyDartClass extends MyClass {
  @override
  int foo() => 77;
}
```

This works but forces the user to always override `foo()` , as it's not implemented in `MyClass.dart`. 
If `foo()` was pure virtual in C++ then this would make sense. Is this a technical limitation or just how things are right now ?

Reply:
This is a technical limitation. Not in a sense of this being impossible, but it a sense of this being very complicated. 

Consider: even if we somehow let user to not override `foo()`, we need a way to know that a non-overridden case should call back to C++. So the full infrastructure would be "if it's overridden in Dart, call back to Dart; if it is not, call back to C++". Probably doable, but very complicated. And so far there were no such use cases in the internal projects we use Gluecodium for.

Reply:
Also, constructing such object would be... interesting. Current bindings support either "Dart object backed by C++ object" or "C++ object backed by Dart object". This scenario would need a "C++ object backed by Dart object partially backed by another C++ object".